### PR TITLE
Related Blog Cards

### LINKED ISSUE #17: Blog Related Posts Block

### PR DESCRIPTION
**Pull request by qilto**
- Add related blog cards
- Refactor cards so that they all use the same "add link to card" function

Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #17

Test URLs:
- Before: https://main--uhg-surest--hlxsites.hlx.page/blog/what-to-know-about-surest-when-choosing-health-insurance
- After: https://blog-related--uhg-surest--hlxsites.hlx.page/blog/what-to-know-about-surest-when-choosing-health-insurance

### CODE REVIEW

File: blocks/cards/cards.js [hunk 1]
```diff
@@ -27,7 +89,10 @@ function addLinksToFeaturedCards(block) {
   });
 }
 
-export default function decorate(block) {
+export default async function decorate(block) {
+  if (getMetadata('template') === 'Blog Post') {
```
Comment (shsteimer) [inline blocks/cards/cards.js:93]: this works on blog pages, but won't on the index page.

Instead lets do `if(block.classList.contains('blog')) { ... }`

File: blocks/cards/cards.js [hunk 2]
```diff
@@ -1,22 +1,84 @@
...
+        }
+      }
+    }
+    relatedBlogList.forEach((blog) => {
+      block.appendChild(createBlogCard(blog));
+    });
+  }
+}
+
+function getBlogLimit(block) {
```
Comment (shsteimer) [inline blocks/cards/cards.js:77]: you don't need your own method for this. use `readBlockConfig` from lib-franklin (this is on me, I should have showed that to you before).

Will give you back js object wit the blog config so you can just do:
```
if (block.classList.contains('blog')) {
const cfg = readBlockConfig(block);
getRelatedBlogs(block, cfg.limit);
}
```

File: blocks/cards/cards.js [hunk 3]
```diff
@@ -1,22 +1,84 @@
...
+}
+
+async function getRelatedBlogs(block, limit = 3) {
+  while (block.firstChild) {
+    block.removeChild(block.lastChild);
+  }
+  const tagList = getMetadata('article:tag').split(', ');
+  const resp = await fetch('/query-index.json');
+  if (resp.ok) {
+    const json = await resp.text();
```
Comment (shsteimer) [inline blocks/cards/cards.js:58]: you can use `resp.json()` so you don't need to call JSON.parse

File: blocks/cards/cards.js [hunk 4]
```diff
@@ -1,22 +1,84 @@
...
+    const blogs = JSON.parse(json);
+    const relatedBlogList = [];
+    for (let i = 0; i < blogs.data.length; i += 1) {
+      const tag = tagList.slice(-1)[0];
```
Comment (shsteimer) [inline blocks/cards/cards.js:62]: 😖 

what does this do? I'm confused, but maybe I'm just being dumb

**Review by shsteimer — APPROVED** [review]
a few comments, but nothing I see as a deal breaker that we can't cleanup on the next pass.

File: blocks/cards/cards.js (hunk 4 above)
Comment (qilto) [reply blocks/cards/cards.js:62]: Actually, I'm clicking through some more blogs and that doesn't seem right at all. I'll update this behavior.

**Pull request merged**